main/src/ca/uwaterloo/flix/language/phase/Reducer.scala

```diff
@@ -57,8 +58,15 @@ object Reducer {
     case Def(ann, mod, sym, cparams, fparams, lparams, _, exp, tpe, unboxedType, loc) =>
       implicit val lctx: LocalContext = LocalContext.mk(exp.purity)
       assert(lparams.isEmpty, s"Unexpected def local params before Reducer: $lparams")
+
       val e = visitExpr(exp)
       val ls = lctx.lparams.toList
+
+      var varOffset = 0
+      for (param <- d.cparams ::: d.fparams ::: ls) {
```

Comment:
Turns out, since we collect local variables into `lparams`, we have all variables as a list so we don't even have to assign offsets "online".

Reply:
Cool.